sources/web/datalab/server.ts

```diff
@@ -155,10 +155,15 @@ function handleRequest(request: http.ServerRequest,
       (path.indexOf('/edit') == 0) ||
       (path.indexOf('/sessions') == 0)) {
 
-    if (path.indexOf('/tree') == 0) {
-        loadedSettings[startup_path_setting] = path
-        settings_.updateUserSetting(userId, startup_path_setting, path, true);
+    // When the user asks for something in the tree, we eventually get called
+    // here with a corresponding URL that starts with /api/contents/.
```

Comment:
I think this changes the behavior of this feature. If you look through the debug logs, you'll see that `/api/contents` requests are made periodically by the frontend to poll for status, even if you don't navigate. This means if you have several tabs open with multiple directories (which is often the case for me at least), every tab you activate will at some point send a `/api/contents` request, saving it as startup path. I don't think this behavior is worse, but it's different, we might want to think through which is better for the user experience.

If we want to retain the original behavior, we should stick to triggering off of the `/tree` requests, and use the file system capabilities to check whether there's a directory with this name.